singularity-eos/eos/modifiers/zsplit_eos.hpp

```diff
@@ -224,6 +226,10 @@ class ZSplit : public EosBase<ZSplit<ztype, T>> {
 
   PORTABLE_INLINE_FUNCTION
   int nlambda() const noexcept { return 1 + t_.nlambda(); }
+  template <typename Indexable>
+  static inline constexpr bool NeedsLambda() {
+    return std::is_same<Indexable, IndexableTypes::MeanIonizationState>::value;
```

Comment:
Should this also have `|| T::template NeedsLambda<Indexable>();`?

Reply:
Yes. Good catch!